libgalois/include/katana/GraphTopology.h

```diff
@@ -1272,6 +1272,146 @@ class KATANA_EXPORT BasicBiDirTopoWrapper
 using SimpleBiDirTopology =
     BasicBiDirTopoWrapper<GraphTopology, EdgeShuffleTopology>;
 
+template <typename OutTopo, typename InTopo>
+class KATANA_EXPORT UndirectedTopologyImpl : public GraphTopologyTypes {
+  // Important:
+  // We assign fake Edge IDs to in_edges to separate them from out edges
+  // fake in-edge-ID == real in-edge-ID + out().num_edges();
```

Comment:
This is a good idea. Do we do the same for bi-directional view as well?

Reply:
No, we don't, since `in_edges() & in_edge_dest()` are separate. But we could use it as added security.